### PR TITLE
Add translation endpoint

### PR DESCRIPTION
**Pull request by erral**
See https://github.com/plone/plone.restapi/issues/225#issuecomment-298526226 for rational behind this.

### CODE REVIEW

**Comment by tisto** [comment]
@erral is this pr ready for review? If so please remove the "in progress" tag.

**Comment by erral** [comment]
Yes, sorry, I was expecting some comments on #225 that's why I didn't remove the in-progress tag, but it's ready for review anyway.

File: CHANGES.rst [hunk 1]
```diff
@@ -18,7 +21,7 @@ New Features:
 
 New Features:
 
-- Add @history endpoint.
+7- Add @history endpoint.
```
Comment (gforcada) [inline CHANGES.rst]: Seems a typo?

Comment (erral) [inline CHANGES.rst]: Indeed. Fixed.

File: docs/source/_json/translations_delete.req [hunk 2]
```diff
@@ -0,0 +1,6 @@
+DELETE /plone/en/test-document/@translations HTTP/1.1
+Accept: application/json
+Authorization: Basic YWRtaW46c2VjcmV0
+Content-Type: application/json
+
+{"language": "es"}
```
Comment (tisto) [inline docs/source/_json/translations_delete.req]: @erral cmon, you could at least use Basque or Catalan. ;)

**Comment by tisto** [comment]
@erral sorry to keep you waiting with this pr! @sneridagh would you mind having a look? You have far more experience with this.

**Comment by erral** [comment]
Some weeks ago I worked on this again.

I limited the `@translations` endpoint to be just Plone 5 compatible. The reason behind this is that Plone 4 doesn't have out of the box multilingual-content support: one needs to install LinguaPlone or p.a.m in those versions.

Moreover, the existing ecosystem of different versions of p.a.m for Plone 4 (1.x and 2.x), require different test setups which I was unable to configure. 

I think that it's better to provide only support for something that it's directly supported on plain Plone.

Moreover I needed some multilingual content support for one of our clients in Plone 4 and LinguaPlone, and extracted this endpoint to 2 dedicated add-ons that just support Plone 4 with LinguaPlone and Plone 4 with p.a.m:

https://github.com/collective/collective.restapi.linguaplone
https://github.com/collective/collective.restapi.pam

With this 2 addons and the support for Plone 5 in this pull request, I think that the multilingual-content support should be complete.

File: docs/source/translations.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
+Translations
+============
+
+.. note::
+    This is only available on Plone 5.
+
+
```
Comment (tisto) [inline docs/source/translations.rst]: @erral a short introduction about Plone's multilingual capabilities would be nice. Keep in mind that the docs should work for JS devs who do not know the internals of Plone. One or two sentences are sufficient.

**Comment by tisto** [comment]
@sneridagh would you mind having a quick look at this?

**Pull request merged**